pageserver/src/tenant/mgr.rs

```diff
@@ -1440,6 +1440,31 @@ impl TenantManager {
         tenant_shard_id: TenantShardId,
         new_shard_count: ShardCount,
         ctx: &RequestContext,
+    ) -> anyhow::Result<Vec<TenantShardId>> {
+        let r = self
+            .do_shard_split(tenant_shard_id, new_shard_count, ctx)
+            .await;
+        if r.is_err() {
+            // Shard splitting might have left the original shard in a partially shut down state (it
+            // stops the shard's remote timeline client).  Reset it to ensure we leave things in
+            // a working state.
+            if self.get(tenant_shard_id).is_some() {
+                tracing::warn!("Resetting {tenant_shard_id} after shard split failure");
+                if let Err(e) = self.reset_tenant(tenant_shard_id, false, ctx).await {
+                    // Log this error because our return value will still be the original error, not this one.
+                    tracing::warn!("Failed to reset {tenant_shard_id}: {e}");
+                }
```

Comment:
Presumably because we expect the storage controller to clean up

Reply:
Yeah.  The ugly reason for dropping out here is just that there's nothing we can do: if reset doesn't work, then it's either something harmless (like the tenant was externally detached/split between our check and this line), or something's bugged and shutdown probably won't work either.

I've extended the comment in 6dad5aaf2 to explain that this isn't really an okay thing to happen, but that we proceed because there's nothing else we can do.